tests/mpi/create_mpi_datatype_01.cc

```diff
@@ -0,0 +1,116 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Test Utilities::MPI::create_mpi_data_type_n_bytes
+
+// To not require a lot of memory, we run this test with smaller
+// size. Change this bool to test the real 64bit size communication:
+const bool run_big = false;
+
+#include <deal.II/base/mpi.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+void
+test_data_type(long long n_bytes)
```

Comment:
```suggestion
test_data_type(const std::uint64_t n_bytes)
```

Reply:
ok. what is your preference for the argument ofcreate_mpi_data_type_n_bytes()? MPI_Count, size_t, std::uint64_t?

Reply:
The whole point of this function is to work around limitations in MPI data type sizes :-) So let's do `std::uint64_t`.

Reply:
Or `std::size_t`. Maybe that's the better choice because that's always going to be big enough to address all elements of any array you can think of.